crates/storage/src/repository.rs

```diff
@@ -29,6 +30,17 @@ use crate::{
     },
 };
 
+/// A [`RepositoryFactory`] is a factory that can create a [`BoxRepository`]
+// XXX(quenting): this could be generic over the repository type, but it's annoying to make it dyn-safe
```

Comment:
meh, don't overcomplicate it any more than it needs to be. For this, I don't think we should worry about the boxing